packages/language-server/src/lib/ast/block.ts

```diff
@@ -0,0 +1,232 @@
+import { Position, Range } from 'vscode-languageserver'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import { MAX_SAFE_VALUE_i32 } from '../types'
+import { getFieldType } from './fields'
+import { getBlockAtPosition } from './findAtPosition'
+
+export type BlockType = 'generator' | 'datasource' | 'model' | 'type' | 'enum' | 'view'
```

Comment:
I'm not sure how we want to organize types, maybe this could go in `../types`? What do you think?

Reply:
Yeah, absolutely!